From 68861a536b256231a395ad9476e95f347d0b0326 Mon Sep 17 00:00:00 2001
From: Enric Balletbo i Serra <eballetbo@iseebcn.com>
Date: Mon, 20 Jun 2011 13:27:17 +0200
Subject: [PATCH 9/9] ASoC: TWL4030: Disable REG_VOICE_IF by default

On IGEP boards, we need to set the TWL4030 PCM voice interface output
pins to high impedance, otherwise this module forces the level of
mcbsp3_fsx and mcbsp3_clkx pins, then the modem connected to uart2 on
IGEP boards doesn't work. As a workaround this patch disables the
REG_VOICE_IF by default.

PCM/codec/TDM/Bluetooth interface modes cannot be changed on the fly,
so we need to set the value 0x04 to TWL4030_REG_VOICE_IF at this file
	 -       0x00,  REG_VOICE_IF  (0xF)
	 +       0x04,  REG_VOICE_IF  (0xF)

Signed-off-by: Enric Balletbo i Serra <eballetbo@iseebcn.com>
---
 sound/soc/codecs/twl4030.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/sound/soc/codecs/twl4030.c b/sound/soc/codecs/twl4030.c
index f798247..b218181 100644
--- a/sound/soc/codecs/twl4030.c
+++ b/sound/soc/codecs/twl4030.c
@@ -65,7 +65,7 @@ static const u8 twl4030_reg[TWL4030_CACHEREGNUM] = {
 	0x0f, /* REG_AVTXL2PGA		(0xC)	*/
 	0x0f, /* REG_AVTXR2PGA		(0xD)	*/
 	0x00, /* REG_AUDIO_IF		(0xE)	*/
-	0x00, /* REG_VOICE_IF		(0xF)	*/
+	0x04, /* REG_VOICE_IF		(0xF)	*/
 	0x3f, /* REG_ARXR1PGA		(0x10)	*/
 	0x3f, /* REG_ARXL1PGA		(0x11)	*/
 	0x3f, /* REG_ARXR2PGA		(0x12)	*/
-- 
1.7.5.4

